app/jobs/two_factor_authentication_removal_job.rb

```diff
@@ -0,0 +1,36 @@
+class TwoFactorAuthenticationRemovalJob < ApplicationJob
+  queue_as :low
+
+  def perform
+    two_factor_authentication_removals = TwoFactorAuthenticationRemoval.all
+    two_factor_authentication_removals.each do |two_factor_authentication_removal|
+      publisher = Publisher.find(two_factor_authentication_removal.publisher_id)
+
+      # First waiting period of the job - 2 weeks
+      if !two_factor_authentication_removal.two_factor_authentication_removal_time_completed?
+        MailerServices::TwoFactorAuthenticationRemovalReminderEmailer.new(publisher: publisher).perform
+
+      # First waiting period completed - 14th Day
+      elsif two_factor_authentication_removal.two_factor_authentication_removal_time_completed? && !two_factor_authentication_removal.removal_completed?
+        # Remove 2fa registrations, remove channels, disconnect wallet, and put publisher into LOCKED state.
+        ActiveRecord::Base.transaction do
+          publisher.totp_registration.destroy if publisher.totp_registration.present?
+          publisher.u2f_registrations.destroy_all if !publisher.u2f_registrations.empty?
+          publisher.channels.destroy_all if !publisher.channels.empty?
```

Comment:
Actually would you run `DeletePublisherChannelJob`? This makes a call to promo server to remove the unused channel.